rwasm/testdata/tinygo-db/main.go

```diff
@@ -0,0 +1,64 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/suborbital/reactr/api/tinygo/runnable"
+	"github.com/suborbital/reactr/api/tinygo/runnable/db"
+	"github.com/suborbital/reactr/api/tinygo/runnable/db/query"
+	"github.com/suborbital/reactr/api/tinygo/runnable/errors"
+	"github.com/suborbital/reactr/api/tinygo/runnable/log"
+)
+
+type TinygoDb struct{}
+
+func (h TinygoDb) Run(input []byte) ([]byte, error) {
+	// `uuid.Generate().String()` doesn't work in old versions of TinyGo so
+	// this will have to do: https://xkcd.com/221/
+	uuidArg := query.NewArgument("uuid", "d7e00ff6-1b30-48e9-aa7d-dd3db34cb8b5")
+
+	_, err := db.Insert("PGInsertUser",
+		uuidArg,
+		query.NewArgument("email", "connor@suborbital.dev"))
+
+	if err != nil {
+		return nil, errors.WithCode(err, 500)
+	} else {
+		log.Info("insert successful")
+	}
+
+	if result, err := db.Update("PGUpdateUserWithUUID", uuidArg); err != nil {
+		return nil, errors.WithCode(err, 500)
+	} else {
+		log.Info(fmt.Sprintf("update: %s", string(result)))
+	}
+
+	if result, err := db.Select("PGSelectUserWithUUID", uuidArg); err != nil {
+		return nil, errors.WithCode(err, 500)
+	} else {
+		log.Info(fmt.Sprintf("select: %s", string(result)))
+	}
+
+	if result, err := db.Delete("PGDeleteUserWithUUID", uuidArg); err != nil {
+		return nil, errors.WithCode(err, 500)
+	} else {
+		log.Info(fmt.Sprintf("delete: %s", string(result)))
+	}
+
+	// this one should fail
+	if result, err := db.Select("PGSelectUserWithUUID", uuidArg); err != nil {
+		return nil, errors.WithCode(err, 500)
```

Comment:
```suggestion
		return nil, errors.WithCode(err, http.StatusInternalServerError)
```

Reply:
We use 500, but it isn't _really_ an HTTP status code. Also, I'm fairly certain we can't import the `http` package from inside of tinygo in wasi because of a `net` dependency.

What we should really do is define our own standard codes for runtime errors.

Reply:
Yes, and we'll need to add and synchronize those same values across all of the other languages and Reactr itself.

Reply:
Oh, hm... this might need rethinking. Let's have a show and tell on how this works, how it should be working, and why the sync is needed. I feel like I'm missing critical need to know information

https://www.youtube.com/watch?v=7QvhCDsFkk0